From d1c25f979062c63bdd80725d13629229fc7ac6b9 Mon Sep 17 00:00:00 2001
From: Chelsio Cudbg <chelsiocudbg@users.noreply.github.com>
Date: Wed, 25 Feb 2026 10:53:43 +0530
Subject: [PATCH 08/13] Revert cxgb4: request the TX CIDX updates to status
 page

commit 7c3bebc3d8688b84795c11848c314a2fbfe045e0
Author: Raju Rangoju <rajur@chelsio.com>
Date:   Wed Oct 23 23:03:55 2019 +0530

    cxgb4: request the TX CIDX updates to status page

    For adapters which support the SGE Doorbell Queue Timer facility,
    we configured the Ethernet TX Queues to send CIDX Updates to the
    Associated Ethernet RX Response Queue with CPL_SGE_EGR_UPDATE
    messages to allow us to respond more quickly to the CIDX Updates.
    But, this was adding load to PCIe Link RX bandwidth and,
    potentially, resulting in higher CPU Interrupt load.

    This patch requests the HW to deliver the CIDX updates to the TX
    queue status page rather than generating an ingress queue message
    (as an interrupt). With this patch, the load on RX bandwidth is
    reduced and a substantial improvement in BW is noticed at lower
    IO sizes.

    Fixes: d429005fdf2c ("cxgb4/cxgb4vf: Add support for SGE doorbell queue timer")
    Signed-off-by: Raju Rangoju <rajur@chelsio.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/chelsio/cxgb4/sge.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb4/sge.c b/drivers/net/ethernet/chelsio/cxgb4/sge.c
index 5fab621efb87..488945da7e6b 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/sge.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/sge.c
@@ -4492,13 +4492,14 @@ int t4_sge_alloc_eth_txq(struct adapter *adap, struct sge_eth_txq *txq,
 	 * write the CIDX Updates into the Status Page at the end of the
 	 * TX Queue.
 	 */
-	c.autoequiqe_to_viid = htonl(((chip_ver <= CHELSIO_T5) ?
+	c.autoequiqe_to_viid = htonl(((dbqt || chip_ver <= CHELSIO_T5) ?
 				      FW_EQ_ETH_CMD_AUTOEQUIQE_F :
 				      FW_EQ_ETH_CMD_AUTOEQUEQE_F) |
 				     FW_EQ_ETH_CMD_VIID_V(pi->viid));
 
 	c.fetchszm_to_iqid =
-		htonl(FW_EQ_ETH_CMD_HOSTFCMODE_V((chip_ver <= CHELSIO_T5) ?
+		htonl(FW_EQ_ETH_CMD_HOSTFCMODE_V((dbqt ||
+						  chip_ver <= CHELSIO_T5) ?
 						 HOSTFCMODE_INGRESS_QUEUE_X :
 						 HOSTFCMODE_STATUS_PAGE_X) |
 		      FW_EQ_ETH_CMD_PCIECHN_V(cxgb4_port_chan(dev)) |
-- 
2.39.1

